### ISSUE TITLE
Support conda-forge pre-release channels

### ISSUE THREAD

**Issue by beenje** [mamba-org/quetz#399]
conda-forge makes it possible to publish pre-release builds to a specific channel (see https://conda-forge.org/docs/maintainer/knowledge_base.html#creating-a-pre-release-build). The channels are named `conda-forge/label/<package_name>_rc` or `conda-forge/label/<package_name>_dev`.

I'd like to proxy some of those channels (i.e. `conda-forge/label/sardana_dev`).

1. Underscore isn't allowed in the channel name. This is an issue to keep the same name as conda-forge.
2. I tried to define `conda-forge/label/sardana-dev` but it looks like using slash isn't well supported. Trying to delete that channel raises a 405.
3. Defining `sardana-dev` to proxy `https://conda.anaconda.org/conda-forge/label/sardana_dev` doesn't work either. Trying to get the repodata raises 500.

**Comment by stevenryoung** [comment]
I think you are conflating some terms. conda-forge is a channel. I think what you are referring to here are labels: https://docs.anaconda.com/anacondaorg/user-guide/tutorials/#using-labels-in-the-development-cycle

They do seem to behave like channels under the hood, but I think the expected default behavior for Quetz would likely be that when I proxy/mirror a channel that all labels are proxied/mirrored, not that I would need to set up a channel for each label. Also, note that every channel has labels, the default is just "main" (example https://anaconda.org/conda-forge/nifty/labels)

**Comment by wolfv** [comment]
Hi both, yes, indeed, that is something we haven't yet figured out / tackled. 

I am not sure we want to completely follow the "labels" of anaconda.org, but having a way to get & mirror pre-releases would be nice.

Will think about this.

**Comment by beenje** [comment]
> I think you are conflating some terms. conda-forge is a channel. I think what you are referring to here are labels: https://docs.anaconda.com/anacondaorg/user-guide/tutorials/#using-labels-in-the-development-cycle
> 

You are correct, but my understanding is that [labels](https://docs.anaconda.com/anacondaorg/glossary/#label) is something that anaconda decided to implement on their server. It's not something that conda (the client) knows about.
So technically, a label is just another conda channel. And yes, the default channel on anaconda.org is kind of an alias for the main label.

There are other conda repositories solutions, like JFrog Artifactory that don't have the concept of labels.
